backend/src/api/album/get_filters.rs

```diff
@@ -0,0 +1,238 @@
+use anyhow::Context;
+use axum::{extract::Query, Extension, Json};
+use itertools::Itertools;
+use serde::{de::DeserializeOwned, Serialize};
+use serde_rusqlite::from_row;
+
+use std::sync::Arc;
+
+use crate::api::{auth::Authorize, error::Error};
+use crate::{AppState, DbInteractable, SqliteDatabase};
+
+use super::{AlbumFilters, Timeframe};
+
+#[derive(Debug, Default, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub(super) struct AvailableAlbumFilters {
+    authors: Vec<String>,
+    timeframes: Vec<Timeframe>,
+    has_drafts: bool,
+}
+
+// This function returns available filters.
+// If filters are sent in it returns the available filters that can be used alternatively.
+// As an example, the returned authors are all authors that can be found
+// with all other filters applied. If the author is filtered on that is ignored for the
+// returned author since otherwise it would only return one value.
+// The same logic applies to all other filters.
+pub(super) async fn get<D: SqliteDatabase>(
+    Authorize(username): Authorize,
+    Query(filter): Query<AlbumFilters>,
+    Extension(state): Extension<Arc<AppState<D>>>,
+) -> Result<Json<AvailableAlbumFilters>, Error> {
+    let conn = state.pool.get().await.context("Failed to get connection")?;
+
+    conn.interact(move |conn| {
+        let mut authors: Vec<String> = get_filtered_values(
+            conn,
+            "author",
+            AlbumFilters {
+                author: None,
+                ..filter
+            },
+            username.clone(),
+        )?;
+
+        authors.dedup();
```

Comment:
Unless you are returning everything sorted, you will probably still get duplicates since dedup only works on continuous runs of elements in a vec. eg. `[1, 1, 2, 2, 3, 1, 1, 4, 4].dedup() == [1, 2, 3, 1, 4]`

Reply:
I should have read the documentation. :p